substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -487,35 +487,61 @@ where
 			"fatp::ready_at_light"
 		);
 
-		let Ok(block_number) = self.api.resolve_block_number(at) else {
+		// Return an empty ready transactions set if we can not find the number of `at` block hash.
+		let Ok(at_number) = self.api.resolve_block_number(at) else {
 			return Box::new(std::iter::empty())
 		};
 
-		let best_result = {
-			api.tree_route(self.enactment_state.lock().recent_finalized_block(), at).map(
-				|tree_route| {
-					if let Some((index, view)) =
-						tree_route.enacted().iter().enumerate().rev().skip(1).find_map(|(i, b)| {
-							self.view_store.get_view_at(b.hash, true).map(|(view, _)| (i, view))
-						}) {
-						let e = tree_route.enacted()[index..].to_vec();
-						(TreeRoute::new(e, 0).ok(), Some(view))
-					} else {
-						(None, None)
-					}
-				},
-			)
+		let mut to_process = Vec::new();
+		let mut best_view = None;
+		let mut tmp_at_number = at_number;
+		let mut tmp_at = at;
+
+		// Return an empty ready transaction set if we can't find a finalized block to report to,
+		// given we can not look for a view that can provide an approximate ready transaction set
+		// on the fork ending with `at` up to the finalized block.
+		let last_finalized = self.enactment_state.lock().recent_finalized_block();
+		let Ok(last_finalized_number) = self.api.resolve_block_number(last_finalized) else {
+			return Box::new(std::iter::empty())
```

Comment:
Should we return the ready txs from the most recent view though?

Reply:
Yeah, that would make sense.

If we can't resolve block number - it is kinda similar to not finding the tree_route. So `best_result` in old code would be `Err`.

Reply:
Maybe we should set best_view to `None` and just follow the execution path that is below?
Then we would handle fallback in one place - flow would be easier to follow.

Reply:
Could be moved to helper function- for better readablity.